app/models/aws_s3.rb

```diff
@@ -8,33 +8,54 @@
 class AwsS3
   attr_accessor :region, :bucket_name, :access_key_id, :secret_access_key, :client
   def initialize(
-    region:,
+    region: nil,
     bucket_name:,
     access_key_id: nil,
     secret_access_key: nil
   )
     @bucket_name = bucket_name
 
+    region ||= ENV.fetch('AWS_REGION', 'us-east-1')
+
     # if environment is set up right, this can all be:
     # self.client = Aws::S3::Client.new
-    if secret_access_key.present? && secret_access_key != 'unknown'
-      self.client = Aws::S3::Client.new({
+    if ENV['USE_MINIO_ENDPOINT'] == 'true' && ENV['MINIO_ENDPOINT'].present?
+      access_key_id = ENV['AWS_ACCESS_KEY_ID'] unless access_key_id.present?
+      secret_access_key = ENV['AWS_SECRET_ACCESS_KEY'] unless secret_access_key.present?
```

Comment:
Thanks for all this!  I guess we haven't done any local build against Minio yet, this will be super helpful.